### ISSUE TITLE
Question about GT rasterization agreement percentage

### ISSUE THREAD

**Issue by zyuanbing** [mlpc-ucsd/BoundaryFormer#2]
Excellent work! Thanks for the release.
When I ran the [test file](https://github.com/mlpc-ucsd/BoundaryFormer/blob/boundary_former/projects/BoundaryFormer/run-rasterizer-tests.py) after compiling the [diff ras](https://github.com/mlpc-ucsd/BoundaryFormer/tree/boundary_former/projects/BoundaryFormer/boundary_former/layers/diff_ras), I found the GT rasterization agreement percentage is around 60, while the masks produced by [Pytorch version](https://github.com/mlpc-ucsd/BoundaryFormer/blob/60f580baeefb4eefed612b60528b6ca79dcc6089/projects/BoundaryFormer/boundary_former/layers/diff_ras/polygon.py#L111) and [CUDA version](https://github.com/mlpc-ucsd/BoundaryFormer/blob/60f580baeefb4eefed612b60528b6ca79dcc6089/projects/BoundaryFormer/boundary_former/layers/diff_ras/polygon.py#L60) are almost same. And the whole training process works fine. 
```
GT Rasterization agreement: 0.594480990116706
Rasterized agreement (tau = 1.0): 0.9998779296875
Gradient agreement (tau = 1.0): 1.0
```
https://github.com/mlpc-ucsd/BoundaryFormer/blob/60f580baeefb4eefed612b60528b6ca79dcc6089/projects/BoundaryFormer/run-rasterizer-tests.py#L53-L59

Does this implicates there is a gap between the mask produced by HARD_CUDA_RASTERIZER and the gt mask ? And what causes this gap?
Thanks in advance.

**Comment by XJKunnn** [comment]
Hi, I think this may caused by the different values of background pixels. The value of background pixel in `ground_truth_masks` is 0, however it is -1 in `ground_truth_rasterized`.
